File: 2529_maximum-count-of-positive-integer-and-negative-integer/maximum-count-of-positive-integer-and-negative-integer.c
int maximumCount(int* nums, int numsSize){
    int ans=0;
    if(nums[0]>0||nums[numsSize-1]<0){
        ans=numsSize; 
    }else{
        int start=0,end=numsSize-1,n=0,p=0;
        while(start<=end){
            if(nums[start]<0){
                n++;
                start++;
            }
            if(nums[end]>0){
                p++;
                end--;
            }
            if(nums[start]==0&&nums[end]==0){
                break;
            }
        }
        ans=n>p?n:p;
    }
    return ans;
}

File: 2529_maximum-count-of-positive-integer-and-negative-integer/maximum-count-of-positive-integer-and-negative-integer.java
class Solution {
    public int maximumCount(int[] nums) {
        int neg = 0, pos = 0;
        
        // Count negative numbers
        for (int num : nums) {
            if (num < 0) {
                neg++;
            } else if (num > 0) {
                pos++;
            }
        }

        // Return the maximum count
        return Math.max(neg, pos);
    }
}